Userland/Libraries/LibC/arpa/inet.cpp

```diff
@@ -13,49 +14,68 @@ extern "C" {
 
 const char* inet_ntop(int af, const void* src, char* dst, socklen_t len)
 {
-    if (af != AF_INET) {
-        errno = EAFNOSUPPORT;
-        return nullptr;
+    if (af == AF_INET) {
+        if (len < INET_ADDRSTRLEN) {
+            errno = ENOSPC;
+            return nullptr;
+        }
+        auto* bytes = (const unsigned char*)src;
+        snprintf(dst, len, "%u.%u.%u.%u", bytes[0], bytes[1], bytes[2], bytes[3]);
+        return (const char*)dst;
+    } else if (af == AF_INET6) {
+        if (len < INET6_ADDRSTRLEN) {
+            errno = ENOSPC;
+            return nullptr;
+        }
+        auto str = IPv6Address(((in6_addr const*)src)->s6_addr).to_string();
+        if (!str.copy_characters_to_buffer(dst, len)) {
+            errno = ENOSPC;
+            return nullptr;
+        }
+        return (const char*)dst;
     }
-    if (len < 4) {
-        errno = ENOSPC;
-        return nullptr;
-    }
-    auto* bytes = (const unsigned char*)src;
-    snprintf(dst, len, "%u.%u.%u.%u", bytes[0], bytes[1], bytes[2], bytes[3]);
-    return (const char*)dst;
+
+    errno = EAFNOSUPPORT;
+    return nullptr;
 }
 
 int inet_pton(int af, const char* src, void* dst)
 {
-    if (af != AF_INET) {
-        errno = EAFNOSUPPORT;
-        return -1;
-    }
-    unsigned a;
-    unsigned b;
-    unsigned c;
-    unsigned d;
-    int count = sscanf(src, "%u.%u.%u.%u", &a, &b, &c, &d);
-    if (count != 4) {
-        errno = EINVAL;
-        return 0;
+    if (af == AF_INET) {
+        unsigned a, b, c, d;
+        int count = sscanf(src, "%u.%u.%u.%u", &a, &b, &c, &d);
```

Comment:
This looks good, but how about adding another commit replacing the ipv4 part to use AK::IPv4Address for consistency?

Reply:
Yeah sounds like something we should eventually do. Maybe I'll do that in a separate PR some other day.